include/bluetooth/bluetooth.h

```diff
@@ -497,6 +503,26 @@ int bt_le_scan_start(const struct bt_le_scan_param *param, bt_le_scan_cb_t cb);
  */
 int bt_le_scan_stop(void);
 
+/** @brief Register a scanner packet listener.
+ *
+ *  Adds the listener to the list of scanner packet callbacks. The listeners
+ *  are called on every received scanner packet, regardless of what initiated
+ *  the scan.
+ *
+ *  @param listener Listener to register. Must point to static memory.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_scan_listener_add(struct bt_le_scan_listener *listener);
+
+/** @brief Deregister a scanner packet listener.
+ *
+ *  @param listener Listener to remove from the list.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_scan_listener_remove(struct bt_le_scan_listener *listener);
```

Comment:
Mainly a question about API completeness/symmetry. But we don't have a way to unregister callbacks for connection callbacks. Should we keep this in sync? I.e either not have a remove for scan callbacks, or add for connection callbacks.